src/Tasks/GenerateResource.cs

```diff
@@ -2905,12 +2905,17 @@ private void ReadResources(String filename, bool shouldUseSourcePath, String out
                         ReadResources(reader, resXReader, filename);
                         break;
 #else
-                        XDocument doc = XDocument.Load(filename, LoadOptions.PreserveWhitespace);
-                        foreach (XElement dataElem in doc.Element("root").Elements("data"))
+
+                        using (var xmlReader = new XmlTextReader(filename))
                         {
-                            string name = dataElem.Attribute("name").Value;
-                            string value = dataElem.Element("value").Value;
-                            AddResource(reader, name, value, filename);
+                            xmlReader.WhitespaceHandling = WhitespaceHandling.None;
+                            XDocument doc = XDocument.Load(xmlReader, LoadOptions.PreserveWhitespace);
+                            foreach (XElement dataElem in doc.Element("root").Elements("data"))
+                            {
+                                string name = dataElem.Attribute("name").Value;
+                                string value = dataElem.Element("value").Value;
+                                AddResource(reader, name, value, filename);
+                            }
                         }
```

Comment:
Review Note: Prod code change

Reply:
It is related.

With the toolset changes, tests are run for both full Framework and .NET Core no matter which version of MSBuild you're using as the build engine for the build.  I was hitting failures when using .NET Core MSBuild, as it would change the newline characters used in some of MSBuild's resources, which would then cause a test to fail that was checking to make sure the messages didn't have any lines longer than a certain length (80 characters I think).